### PR TITLE
ADB to USB converter based on TMK

### PR DESCRIPTION
**Pull request by KauyonKais**
Title says it all. I adjusted the TMK adb_usb converter a little bit so it works with qmk. I only tested it on a German ISO AEK2 using a ProMicro, but I am confident it will work on the other configurations as expected.  
I got rid of the daisy-chained mouse support for now though.

### CODE REVIEW

File: keyboards/converter/adb_usb/matrix.c [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    for (uint8_t i=0; i < MATRIX_ROWS; i++) matrix[i] = 0x00;
+
+    debug_enable = true;
+    //debug_matrix = true;
+    //debug_keyboard = true;
+    //debug_mouse = true;
+    print("debug enabled.\n");
+
+    // LED off
+    return;
```
[the remark this answers is not in the document]
Comment (KauyonKais) [reply keyboards/converter/adb_usb/matrix.c]: Just so I understand this correctly, `matrix_init_quantum()` is to run at the end of `matrix_init()`?

[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/converter/adb_usb/matrix.c]: Yup.  This calls all the "quantum" stuff, as all as the kb/user versions

File: keyboards/converter/adb_usb/matrix.c [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    if (codes == 0) {                           // no keys
+        return 0;
+    } else if (codes == 0x7F7F) {   // power key press
+        register_key(0x7F);
+    } else if (codes == 0xFFFF) {   // power key release
+        register_key(0xFF);
+    } else if (key0 == 0xFF) {      // error
+        xprintf("adb_host_kbd_recv: ERROR(%d)\n", codes);
+        // something wrong or plug-in
+        matrix_init();
```
Comment (drashna) [inline keyboards/converter/adb_usb/matrix.c]: Also, this isn't needed here, IIRC.  

The keyboard code calls the init and scan, so this is redundant

**Comment by mechmerlin** [comment]
I'm looking at this from the point of view of the [QMK Configurator](https://config.qmk.fm)

With the current code, a user will choose `converter` in the keyboard dropdown, and then choose one of the layouts in the layout dropdown. This will change the keyboard from an Apple Standard, Apple Extended, MO116 and MO118. I don't think a layout should change the actual physical keyboard. 

It would be nice to refactor this to pull out each of those keyboards into its own directory so that in the keyboard dropdown people choose
converter/mo116
converter/mo118
converter/apple_standard
etc etc

and in the layout dropdown they pick
LAYOUT_iso
LAYOUT_ansi
LAYOUT_whatever

**Comment by mechmerlin** [comment]
@drashna I was under the impression that the converter was a adb to usb converter. So it seems rather repetitive to do that. Maybe one day when we have more converts we can.

**Comment by drashna** [comment]
@mechmerlin there is already a USB to USB and ibm terminal converter present:
https://github.com/qmk/qmk_firmware/tree/master/keyboards/converter

**Comment by drashna** [comment]
Looks good to me! 

@jackhumbert this does touch some of the core files, so if you wanted to double check it?

File: keyboards/converter/adb_usb/rules.mk [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+CONSOLE_ENABLE		= yes	# Console for debug(+400)
+COMMAND_ENABLE		= no  # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes	# USB Nkey Rollover - not yet supported in LUFA
+EXTRAKEY_ENABLE		= yes	
+USB_HID_ENABLE = yes
+BACKLIGHT_ENABLE = no
+
+CUSTOM_MATRIX = yes
+SRC = matrix.c
```
Comment (drashna) [inline keyboards/converter/adb_usb/rules.mk]: This should be a `+=`

**Comment by lbibass** [comment]
Just wondering what the state of this was, as I would love to have this in the main QMK files.

File: .gitignore [hunk 4]
```diff
@@ -55,6 +55,7 @@ util/Win_Check_Output.txt
 .vscode/temp.sql
 .stfolder
 .tags
+*.ini
```
[the remark this answers is not in the document]
Comment (drashna) [reply .gitignore]:
```suggestion
```

File: keyboards/converter/adb_usb/rules.mk (hunk 3 above)
Comment (drashna) [reply keyboards/converter/adb_usb/rules.mk]:
```suggestion
SRC += matrix.c
```

**Comment by drashna** [comment]
I'm going to close this, as a different implementation was implemented.

**Pull request closed without merging**